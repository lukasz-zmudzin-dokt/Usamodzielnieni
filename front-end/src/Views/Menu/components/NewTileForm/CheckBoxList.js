import React from "react";
import { Row, Form } from "react-bootstrap";

const CheckBoxList = ({ overlay, setOverlay }) => (
  <>
    <Form.Label>Wybierz przenikanie zdjęcia</Form.Label>
    <Row className="mx-1 mb-2 justify-content-center">
      <Form.Check
        inline
        label="Z lewej"
        type="checkbox"
        id="inline-checkbox-left"
        checked={overlay.left}
        onChange={() => setOverlay({ ...overlay, left: !overlay.left })}
      />
      <Form.Check
        inline
        label="Z góry"
        type="checkbox"
        id="inline-checkbox-top"
        checked={overlay.top}
        onChange={() => setOverlay({ ...overlay, top: !overlay.top })}
      />
      <Form.Check
        inline
        label="Z prawej"
        type="checkbox"
        id="inline-checkbox-right"
        checked={overlay.right}
        onChange={() => setOverlay({ ...overlay, right: !overlay.right })}
      />
    </Row>
  </>
);

export default CheckBoxList;
