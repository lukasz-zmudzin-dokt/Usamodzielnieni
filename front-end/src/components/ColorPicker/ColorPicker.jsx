import { SketchPicker } from "react-color";
import React from "react";

const ColorPicker = ({ background, setBackground }) => (
  <SketchPicker
    className="w-100"
    disableAlpha
    color={background}
    onChangeComplete={(color) => setBackground(color.hex)}
    presetColors={[
      "#f3d689",
      "#feb549",
      "#e4a102",
      "#dcd7d7",
      "#ffb37b",
      "#d8a47f",
      "#e6a4a1",
      "#aa8acc",
      "#7469af",
      "#6699cc",
      "#a3b4c5",
      "#79c9a6",
      "#42b4a8",
    ]}
  />
);

export default ColorPicker;
