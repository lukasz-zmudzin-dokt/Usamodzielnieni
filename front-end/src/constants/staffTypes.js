export const staffTypes = {
  VERIFICATION: "staff_verification",
  CV: "staff_cv",
  JOBS: "staff_jobs",
  BLOG_CREATOR: "staff_blog_creator",
  BLOG_MODERATOR: "staff_blog_moderator",
};
