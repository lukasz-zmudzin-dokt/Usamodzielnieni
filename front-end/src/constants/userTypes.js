export const userTypes = {
  STANDARD: "standard",
  STAFF: "staff",
  EMPLOYER: "employer",
};
