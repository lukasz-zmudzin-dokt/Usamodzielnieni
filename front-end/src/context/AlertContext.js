import React, { useState, createContext } from "react";

export const AlertContext = createContext({
  open: false,
  message: "",
  setOpen: () => {},
});

export const AlertProvider = (props) => {
  const [open, setOpen] = useState(false);
  const [message, setMessage] = useState("");
  const [variant, setVariant] = useState("danger") // success
  const data = {
    open,
    message,
    variant,
    changeVisibility: (newOpen) => setOpen(newOpen),
    changeMessage: (newMessage,newVariant = variant) => {
      setOpen(false);
      setMessage(newMessage);
      setVariant(newVariant);
    },
    showAlert:  (newMessage,newVariant) => {
      data.changeMessage(newMessage,newVariant);
      data.changeVisibility(true);
    }
  };
  return <AlertContext.Provider value={data} {...props} />;
};
