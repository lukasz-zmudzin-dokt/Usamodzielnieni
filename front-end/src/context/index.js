export { UserContext, UserProvider } from "context/UserContext";
export { AlertContext, AlertProvider } from "context/AlertContext";
export { ChatContext, ChatProvider } from "context/ChatContext";
export {
  NotificationsContext,
  NotificationsProvider,
} from "context/NotificationsContext";
