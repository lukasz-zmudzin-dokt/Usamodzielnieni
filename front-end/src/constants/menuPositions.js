import { userTypes } from "./userTypes";
import { staffTypes } from "./staffTypes";
import { paths } from "./paths";

export default [       
    {
        name: "Kreator CV",
        path: paths.CV_CREATOR,
        allowed: [userTypes.STANDARD],
        verified: true
    },
    {
        name: "Moje CV",
        path: paths.MY_CVS,
        allowed: [userTypes.STANDARD],
        verified: true
    },
    {
        name: "Akceptacja CV",
        path: paths.CV_APPROVAL,
        allowed: [staffTypes.CV],
        verified: true
    },
    {
        name: "Oferty pracy",
        path: paths.JOB_OFFERS,
        allowed: [userTypes.STANDARD, userTypes.STAFF, userTypes.EMPLOYER]
    },
    {
        name: "Dodaj ofertę",
        path: paths.OFFER_FORM,
        allowed: [userTypes.EMPLOYER],
        verified: true
    },
    {
        name: "Moje oferty",
        path: paths.MY_OFFERS,
        allowed: [userTypes.EMPLOYER],
        verified: true
    },
    {
        name: "Blogi",
        path: paths.BLOG_PAGE,
        allowed: [userTypes.STANDARD, userTypes.STAFF, userTypes.EMPLOYER]
    },
    {
        name: "Dodaj post",
        path: paths.BLOG_FORM,
        allowed: [staffTypes.BLOG_CREATOR],
        verified: true
    },
    {
        name: "Telefony",
        path: paths.CONTACT_PAGE,
        allowed: undefined, //wszyscy są upoważnieni - zalogowani i niezalogowani
    },
    {
        name: "Kroki",
        path: paths.STEPS,
        allowed: undefined //wszyscy są upoważnieni - zalogowani i niezalogowani
    },
    {
        name: "Chat",
        path: paths.CHATS,
        allowed: [userTypes.STANDARD, userTypes.STAFF, userTypes.EMPLOYER],
        verified: true,
    }
];
