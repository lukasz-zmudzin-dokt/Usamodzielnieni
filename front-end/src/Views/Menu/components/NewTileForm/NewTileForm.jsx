import { Alert, Button, Col, Form, Modal, Row } from "react-bootstrap";
import React, {
  Suspense,
  useContext,
  useEffect,
  useRef,
  useState,
} from "react";
import FormGroup from "components/FormGroup";
import { DEFAULT_INPUT } from "constants/other";
import Tile from "../Tile/Tile";
import { AlertContext } from "context/AlertContext";
import { toBase64 } from "utils/converters/fileToBase64";
import CheckBoxList from "./CheckBoxList";
import { approveFileSize } from "utils/approveFile/approveFile";
import proxy from "config/api";
const ColorPicker = React.lazy(() =>
  import("components/ColorPicker/ColorPicker")
);

const fetchPosts = async () => {
  const url = proxy.blog + "blogposts/";
  const headers = {
    "Content-Type": "application/json",
  };

  const res = await fetch(url, { headers, method: "GET" });

  if (res.status === 200) {
    return await res.json().then((data) => mapPosts(data.results));
  } else {
    throw res.status;
  }
};

const addTile = async (token, tile, mode, id) => {
  let url = proxy.menu + "tile/";
  url = mode === "PUT" ? url + id + "/" : url;
  const headers = {
    Authorization: "Token " + token,
    "Content-Type": "application/json",
  };
  const res = await fetch(url, {
    method: mode,
    headers,
    body: JSON.stringify(tile),
  });
  const status = mode === "POST" ? 201 : 200;
  if (res.status === status) {
    return await res.json();
  } else {
    throw await res.json();
  }
};

const postPhoto = async (token, id, photo) => {
  const url = proxy.menu + "tile/" + id + "/photo/";
  const headers = {
    Authorization: "Token " + token,
    //"Content-Type": "application/x-www-form-urlencoded"
  };
  const pic = new FormData();
  pic.append("photo", photo, photo.name);

  const res = await fetch(url, { method: "POST", headers, body: pic });

  if (res.status === 200) {
    return;
  } else {
    throw await res.json();
  }
};

const mapPosts = (posts) => {
  return posts.map((item) => ({
    id: "/blog/blogpost/" + item.id,
    name: item.title + " (Kategoria: " + item.category + ")",
  }));
};

const NewTileForm = ({ show, setShow, user, appendTile, tileData }) => {
  const [validated, setValidated] = useState(false);
  const [title, setTitle] = useState("");
  const [background, setBackground] = useState("#fafafa");
  const [path, setPath] = useState(DEFAULT_INPUT);
  const [photoB64, setPhotoB64] = useState("");
  const [label, setLabel] = useState("");
  const [tileId, setTileId] = useState(undefined);
  const [method, setMethod] = useState("POST");
  const [imgOverFlow, setImgOverflow] = useState({
    left: false,
    top: false,
    right: false,
  });
  const [pathArray, setPathArray] = useState([]);
  const [loading, setLoading] = useState(false);

  let fileInput = useRef(null);
  const alertContext = useRef(useContext(AlertContext));

  useEffect(() => {
    setLoading(true);
    const loadPostList = async () => {
      let res;
      try {
        res = await fetchPosts();
        setPath(res[0].id);
      } catch (e) {
        console.log(e);
        alertContext.current.showAlert(
          "Wystąpił błąd podczas pobierania danych o postach."
        );
        res = [];
      }
      setPathArray(res);
      if (tileData) {
        const { id, title, color, show, imageUrl, destination } = tileData;
        setTileId(id);
        setTitle(title);
        setBackground(color);
        setImgOverflow(show);
        setPhotoB64(imageUrl);
        setPath(destination);
        setMethod("PUT");
        setLabel("Poprzednie zdjęcie");
      }
    };
    loadPostList();
    setLoading(false);
  }, [alertContext, tileData]);

  const onChange = async () => {
    const file = fileInput.current?.files?.[0];
    if (approveFileSize(file)) {
      try {
        const b64 = await toBase64(file);
        setPhotoB64(b64);
        setLabel(file.name);
      } catch (e) {
        alertContext.current.showAlert(
          "Wystąpił błąd podczas wyświetlania wybranego zdjęcia."
        );
      }
    } else {
      alertContext.current.showAlert(
        "Wybrany plik jest za duży. Maksymalna wielkość załącznika to 15 MB."
      );
      fileInput = null;
    }
  };

  const clearInput = () => {
    setTitle("");
    setBackground("#fafafa");
    setPath("");
    setValidated(false);
    setLabel("");
    setPhotoB64("");
    setImgOverflow({ left: false, top: false, right: false });
    setMethod("POST");
    setTileId(undefined);
    setValidated(false);
    setShow(false);
  };

  const handleSubmit = async (e) => {
    e.preventDefault();
    setValidated(true);
    if (e.currentTarget.checkValidity() === false) {
      e.stopPropagation();
    } else {
      const data = {
        title: title,
        destination: path,
        color: background,
        photo_layer: imgOverFlow,
      };
      try {
        const res = await addTile(user.token, data, method, tileId);
        setTileId(res.id);
        !tileData &&
          (await postPhoto(user.token, res.id, fileInput.current.files[0]));
        alertContext.current.showAlert("Kafelek dodany pomyślnie.", "success");
        appendTile({
          id: res.id || tileId,
          title: title,
          color: background,
          show: imgOverFlow,
          imageUrl: photoB64,
          destination: path,
        });
        clearInput();
      } catch (e) {
        console.log(e);
        alertContext.current.showAlert(Object.values(e)[0]);
      }
    }
  };

  return loading ? (
    <Alert variant="info">Ładowanie...</Alert>
  ) : (
    <Modal
      show={show}
      size="lg"
      onHide={(e) => {
        clearInput();
      }}
    >
      <Modal.Header closeButton>
        <Modal.Title>Nowy kafelek</Modal.Title>
      </Modal.Header>
      <Form noValidate validated={validated} onSubmit={handleSubmit}>
        <Modal.Body>
          <FormGroup
            header="Tytuł kafelka"
            setVal={setTitle}
            id="title"
            val={title}
            incorrect="Tytuł kafelka nie może być pusty"
            required
          />
          <Form.Label>Ścieżka do kafelka</Form.Label>
          <Form.Control
            as="select"
            value={path}
            id="path"
            onChange={(v) => setPath(v.target.value)}
            required
          >
            {pathArray.map((val) => (
              <option key={val.id} value={val.id}>
                {val.name}
              </option>
            ))}
          </Form.Control>
          <Form.Label>Zdjęcie kafelka</Form.Label>
          <Form.File
            className="text-nowrap text-truncate"
            id="custom-file"
            ref={fileInput}
            custom
            onChange={onChange}
            label={label || "Dodaj zdjęcie..."}
            accept="image/*"
            data-browse="Wybierz plik"
            required={!tileData}
          />
          <Form.Control.Feedback type="invalid">
            Podaj zdjęcie kafelka
          </Form.Control.Feedback>
          <CheckBoxList overlay={imgOverFlow} setOverlay={setImgOverflow} />
          <Row>
            <Col>
              <Form.Label>Kolor kafelka</Form.Label>
              <Suspense fallback={<Alert variant="info">Ładowanie...</Alert>}>
                <ColorPicker
                  background={background}
                  setBackground={setBackground}
                />
              </Suspense>
            </Col>
            <Col>
              <Form.Label>Podgląd</Form.Label>
              <Tile
                title={title}
                color={background}
                destination={""}
                imageUrl={photoB64}
                showImage={imgOverFlow}
                previewOnly={true}
              />
            </Col>
          </Row>
        </Modal.Body>
        <Modal.Footer>
          <Button type="submit" variant="primary">
            Dodaj kafelek
          </Button>
        </Modal.Footer>
      </Form>
    </Modal>
  );
};

export default NewTileForm;
