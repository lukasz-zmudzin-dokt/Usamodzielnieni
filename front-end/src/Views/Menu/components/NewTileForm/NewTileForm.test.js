import proxy from "config/api";
import ColorPicker from "components/ColorPicker/ColorPicker";
import React from "react";
import { Tile } from "../index";
import { userTypes } from "constants/userTypes";
import { staffTypes } from "constants/staffTypes";
import { AlertContext } from "context/AlertContext";
import NewTileForm from "./NewTileForm";
import { render, fireEvent, wait } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";

jest.mock("../");
//jest.mock("components");

describe("NewTileForm", () => {
  let photoFail,
    categoryFail,
    postFail,
    show,
    setShow,
    user,
    appendTile,
    tileData,
    alertC;
  let props;
  beforeAll(() => {
    //ColorPicker.mockImplementation(() => <div/>);
    Tile.mockImplementation(
      (title, color, destination, imageUrl, showImage) => (
        <div>
          {title} {color} {destination} {imageUrl} {showImage}
        </div>
      )
    );
    global.fetch = jest.fn().mockImplementation((input, init) => {
      return new Promise((resolve, reject) => {
        if (photoFail || postFail || categoryFail) {
          resolve({
            status: 500,
            json: () => Promise.resolve({ message: "Błąd serwera." }),
          });
        } else if (init === "GET") {
          resolve({
            status: 200,
            json: () =>
              Promise.resolve([{ id: 1, category: "cat", title: "t" }]),
          });
        } else {
          switch (input) {
            case proxy.menu + "tile/123/photo":
              resolve({ status: 200, json: () => Promise.resolve("") });
              break;
            case proxy.menu + "tile/":
              if (init.method === "POST") {
                resolve({
                  status: 201,
                  json: () => Promise.resolve({ id: 123 }),
                });
              } else {
                resolve({ status: 200, json: () => Promise.resolve("git") });
              }
              break;
            default:
              reject({});
          }
        }
      });
    });
  });

  beforeEach(() => {
    show = true;
    setShow = jest.fn();
    user = {
      token: "123231",
      type: userTypes.STAFF,
      data: {
        group_type: [staffTypes.BLOG_MODERATOR],
      },
    };
    appendTile = jest.fn();
    tileData = {
      id: 123,
      title: "tytuł",
      color: "#ababab",
      show: { left: true, top: true, right: true },
      imageUrl: "photophotophotophoto",
      destination: "/qwe/qwe/123",
    };
    photoFail = false;
    categoryFail = false;
    postFail = false;
    alertC = {
      showAlert: jest.fn(),
    };
    jest.clearAllMocks();
    props = {
      show: show,
      setShow: setShow,
      user: user,
      appendTile: appendTile,
    };
  });

  it("should match snapshot", () => {
    const { getByRole } = render(
      <AlertContext.Provider value={alertC}>
        <MemoryRouter>
          <NewTileForm {...props} />
        </MemoryRouter>
      </AlertContext.Provider>
    );

    const modal = getByRole("dialog");

    expect(modal).toMatchSnapshot();
  });

  // it('should validate form and append tile', async () => {
  //     const {getByText, getByLabelText} = render(
  //         <AlertContext.Provider value={alertC}>
  //             <MemoryRouter>
  //                 <NewTileForm {...props} />
  //             </MemoryRouter>
  //         </AlertContext.Provider>
  //     );
  //
  //     await expect(fetch).toHaveBeenCalled();
  //
  //     fireEvent.change(getByLabelText("Tytuł kafelka"), {
  //         target: {value: "tytuł"}
  //     });
  //
  //     fireEvent.change(getByLabelText("Ścieżka do kafelka"), {
  //         target: {value: "qwe"}
  //     });
  //
  //     fireEvent.change(getByLabelText("Zdjęcie kafelka", {exact: false}), {
  //         target: {
  //             files: ["dummyValue.something"],
  //         },
  //     });
  //
  //     fireEvent.click(getByText("Dodaj kafelek"));
  //
  //     await wait(() => expect(fetch).toHaveBeenCalled());
  //
  //     expect(alertC.showAlert).toHaveBeenCalledWith(
  //         "Kafelek dodany pomyślnie.", "success"
  //     );
  //
  //     expect(appendTile).toHaveBeenCalledWith({
  //         id: 123,
  //         title: "tytuł",
  //         destination: "qwe",
  //         color: "#fafafa",
  //         show: {left: false, top: false, right: false},
  //         imageUrl: expect.anything()
  //     });
  // });
  //
  // it('should fill form with values on edit', () => {
  //     const {getByText, getByLabelText} = render(
  //         <AlertContext.Provider value={alertC}>
  //             <MemoryRouter>
  //                 <NewTileForm {...props} tileData={tileData}/>
  //             </MemoryRouter>
  //         </AlertContext.Provider>
  //     );
  //
  //     expect(getByLabelText("Tytuł kafelka").value).toBe("tytuł");
  //     expect(getByLabelText("Ścieżka do kafelka").value).toBe("/qwe/qwe/123");
  //     //expect(getByLabelText("Zdjęcie kafelka").files).toBe("photophotophotophoto");
  // });
});
