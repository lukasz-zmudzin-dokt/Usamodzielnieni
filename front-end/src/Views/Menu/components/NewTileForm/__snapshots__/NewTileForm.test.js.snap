// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewTileForm should match snapshot 1`] = `
<div
  aria-modal="true"
  class="fade modal show"
  role="dialog"
  style="display: block;"
  tabindex="-1"
>
  <div
    class="modal-dialog modal-lg"
    role="document"
  >
    <div
      class="modal-content"
    >
      <div
        class="modal-header"
      >
        <div
          class="modal-title h4"
        >
          Nowy kafelek
        </div>
        <button
          class="close"
          type="button"
        >
          <span
            aria-hidden="true"
          >
            ×
          </span>
          <span
            class="sr-only"
          >
            Close
          </span>
        </button>
      </div>
      <form
        class=""
        novalidate=""
      >
        <div
          class="modal-body"
        >
          <div
            class="form-group"
          >
            <label
              class="form-label"
              for="title"
            >
              Tytuł kafelka
            </label>
            <input
              class="form-control"
              data-testid="default"
              id="title"
              maxlength="50"
              minlength="1"
              placeholder="Tytuł kafelka"
              required=""
              type="text"
              value=""
            />
            <div
              class="invalid-feedback"
            >
              Tytuł kafelka nie może być pusty
            </div>
          </div>
          <div
            class="form-group"
          >
            <label
              class="form-label"
              for="path"
            >
              Ścieżka do kafelka
            </label>
            <select
              class="form-control"
              id="path"
              required=""
            />
          </div>
          <div
            class="form-group"
          >
            <label
              class="form-label"
              for="custom-file"
            >
              Zdjęcie kafelka
            </label>
            <div
              class="text-nowrap text-truncate custom custom-file"
            >
              <input
                accept="image/*"
                class="custom-file-input"
                id="custom-file"
                required=""
                type="file"
              />
              <label
                class="custom-file-label"
                data-browse="Wybierz plik"
                for="custom-file"
              >
                Dodaj zdjęcie...
              </label>
            </div>
            <div
              class="invalid-feedback"
            >
              Podaj zdjęcie kafelka
            </div>
          </div>
          <label
            class="form-label"
          >
            Wybierz przenikanie zdjęcia
          </label>
          <div
            class="mx-1 mb-2 justify-content-center row"
          >
            <div
              class="form-check form-check-inline"
            >
              <input
                class="form-check-input"
                id="inline-checkbox-left"
                type="checkbox"
              />
              <label
                class="form-check-label"
                for="inline-checkbox-left"
                title=""
              >
                Z lewej
              </label>
            </div>
            <div
              class="form-check form-check-inline"
            >
              <input
                class="form-check-input"
                id="inline-checkbox-top"
                type="checkbox"
              />
              <label
                class="form-check-label"
                for="inline-checkbox-top"
                title=""
              >
                Z góry
              </label>
            </div>
            <div
              class="form-check form-check-inline"
            >
              <input
                class="form-check-input"
                id="inline-checkbox-right"
                type="checkbox"
              />
              <label
                class="form-check-label"
                for="inline-checkbox-right"
                title=""
              >
                Z prawej
              </label>
            </div>
          </div>
          <div
            class="row"
          >
            <div
              class="col"
            >
              <label
                class="form-label"
              >
                Kolor kafelka
              </label>
              <div
                class="fade alert alert-info show"
                role="alert"
              >
                Ładowanie...
              </div>
            </div>
            <div
              class="col"
            >
              <label
                class="form-label"
              >
                Podgląd
              </label>
              <div
                class="tile__container"
                style="background-color: rgb(116, 105, 175);"
              >
                <a
                  href="/"
                >
                  <div
                    class="tile"
                  >
                    <div
                      class="tile__bg tile__bg--left "
                      style="background-color: rgb(116, 105, 175);"
                    />
                    <div
                      class="tile__bg tile__bg--top "
                      style="background-color: rgb(116, 105, 175);"
                    />
                    <div
                      class="tile__bg tile__bg--right "
                      style="background-color: rgb(116, 105, 175);"
                    />
                    <div
                      class="tile_content"
                    >
                      <div
                        class="tile__border tile__border--left "
                      />
                      <div
                        class="tile__border tile__border--top "
                      />
                      <div
                        class="tile__border tile__border--right "
                      />
                      <div
                        class="tile__image"
                      >
                        <img
                          alt=""
                          src=""
                        />
                      </div>
                      <div
                        class="tile_title"
                      >
                        <span />
                      </div>
                    </div>
                  </div>
                </a>
              </div>
            </div>
          </div>
        </div>
        <div
          class="modal-footer"
        >
          <button
            class="btn btn-primary"
            type="submit"
          >
            Dodaj kafelek
          </button>
        </div>
      </form>
    </div>
  </div>
</div>
`;
